functions/supplier/http-supplier/README.adoc

```diff
@@ -0,0 +1,33 @@
+# HTTP Supplier
+
+This module provides an HTTP supplier that can be reused and composed in other applications.
+The `Supplier` uses the `HttpInboundChannelAdapter` from Spring Integration.
```

Comment:
I think it's better (simpler) to use `link:<relative_path>` rather than githib urls (see the function README in the http client PR).  For one thing, before the PR is merged, the url does not exist on master so must be manually edited and cannot be verified. For another, the syntax is much simpler.